api/v1alpha1/network_types.go

```diff
@@ -27,6 +27,9 @@ type NetworkSpec struct {
 	// +listType=map
 	// +listMapKey=id
 	PeeredPrefixes []PeeredPrefix `json:"peeredPrefixes,omitempty" patchStrategy:"merge" patchMergeKey:"peeredPrefixes"`
+
+	// InternetGateway is a flag that indicates whether the network has an internet gateway.
+	InternetGateway bool `json:"internetGateway,omitempty"`
```

Comment:
> Think twice about bool fields. Many ideas start as boolean but eventually trend towards a small set of mutually exclusive options. Plan for future expansions by describing the policy options explicitly as a string type alias (e.g. TerminationMessagePolicy).

https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#primitive-types

Reply:
OSC dev here: I do not agree with this "plan for future expansions" mindset.
This is why we have API versioning like `v1alpha1` etc.
When this bool field is not enough then just create `v1alpha2` package with refactored stuff. 

If you plan for "future expansions" then you are just commiting unecessary complexity that is not needed right now. And may be it will be not needed at all.

Start simple, refactor later if needed.

Reply:
So what you are suggesting is: instead of starting with a more flexible approach which is in line with the kube API convention, we should rather go down the rabbit hole of API conversion?

Reply:
hmm, TSI currently is struggling a lot with a non-stable API. So you should think about how to design the API in an extensible way, that does not break old integrations, when being upgraded.

Reply:
Just to be clear: This flag will never make its way to the upstream main branch.